processing/src/main/java/io/druid/segment/SingleScanTimeDimSelector.java

```diff
@@ -72,7 +74,7 @@ public IndexedInts getRow()
     // we can also avoid creating a dimension value and corresponding index
     // and use the current one
     else if (timestamp != currentTimestamp) {
-      if(timestamp < currentTimestamp) {
+      if (descending && timestamp > currentTimestamp || !descending && timestamp < currentTimestamp) {
```

Comment:
nit: can you add parentheses bcoz its not obvious if `&&` has higher precedence than `||` ?